nucypher/characters/lawful.py

```diff
@@ -1106,10 +1113,13 @@ def __init__(self,
             self._availability_tracker = AvailabilityTracker(ursula=self)
 
             # Datastore Pruning
-            self.__pruning_task = None
             self._prune_datastore = prune_datastore
             self._datastore_pruning_task = LoopingCall(f=self.__prune_datastore)
 
+            # Known nodes pruning
+            self._node_garbage_collection = node_garbage_collection
+            self._node_pruning_task = LoopingCall(f=self.__prune_nodes)
```

Comment:
So this means that pruning is for Ursulas only, right? Alice could benefit from it too.

Reply:
I agree. Despite Alice running the discovery loop for such a short amount of time it will still be useful, although this will be much more effective it she continues to run node discovery as a damon.